pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -166,7 +175,13 @@ type EndpointController struct {
 	// annotation.
 	triggerTimeTracker *endpointutil.TriggerTimeTracker
 
-	endpointUpdatesBatchPeriod time.Duration
+	// syncRateLimiter is used to track per-service rate limiter.
+	syncRateLimiter *workqueue.ItemBucketRateLimiter
+
+	// queuedNotStarted keeps a set of service keys that has been added to queue as an effect of service/pod changes,
+	// and not started yet.
+	queuedNotStarted     map[string]bool
```

Comment:
nit: how about using strings.Set instead of this map:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/sets/string.go

Reply:
Done